internal/logging/logger.go

```diff
@@ -73,125 +74,145 @@ var (
 		"INFO":    zerolog.InfoLevel,
 		"DEBUG":   zerolog.DebugLevel,
 		"TRACE":   zerolog.TraceLevel,
+		"UNSET":   unset,
 	}
 )
 
 func NewLogger(zerologLogger zerolog.Logger) *Logger {
 	return &Logger{
-		l:                         &zerologLogger,
+		defaultLogger:             &zerologLogger,
 		scopeLoggers:              make(map[string]*zerolog.Logger),
 		scopeLevels:               make(map[string]zerolog.Level),
-		scopeLevelMutex:           sync.Mutex{},
-		defaultLogLevelFromEnv:    -2,
-		defaultLogLevelFromConfig: -2,
+		loggerMutex:               sync.Mutex{},
+		defaultLogLevelFromEnv:    unset,
+		defaultLogLevelFromConfig: unset,
 		defaultLogLevel:           defaultLogLevel,
 	}
 }
 
-func (l *Logger) updateLogLevel() {
-	l.scopeLevelMutex.Lock()
-	defer l.scopeLevelMutex.Unlock()
+func (l *Logger) updateLogLevels(newConfigLevel zerolog.Level) {
+	logger := l.defaultLogger.
+		Level(zerolog.InfoLevel).
+		With().
+		Interface("newConfigLevel", newConfigLevel).
+		Logger()
 
+	l.defaultLogLevelFromConfig = newConfigLevel
 	l.scopeLevels = make(map[string]zerolog.Level)
 
-	finalDefaultLogLevel := l.defaultLogLevel
-
-	for name, level := range zerologLevels {
+	for name, envLevel := range zerologLevels {
 		env := os.Getenv(fmt.Sprintf("JETKVM_LOG_%s", name))
 
-		if env == "" {
-			env = os.Getenv(fmt.Sprintf("PION_LOG_%s", name))
-		}
-
-		if env == "" {
-			env = os.Getenv(fmt.Sprintf("PIONS_LOG_%s", name))
-		}
-
-		if env == "" {
-			continue
+		if env != "" {
+			env = strings.ToLower(env)
+			loopLogger := logger.With().
```

Comment:
... huh, inception... logging out the level setting of the logging infrastructure... weeeeee